src/test/run-pass/ctfe/references.rs

```diff
@@ -28,9 +28,10 @@ fn main() {
         _ => panic!("c"),
     }
 
+    #[allow(unreachable_patterns)]
     match &43 {
         &42 => panic!(),
-        BOO => panic!(),
+        BOO => panic!(), // pattern is unreachable
```

Comment:
This is a behavioral change of this PR. We now detect that `BOO` is `&42`. This solely emits new warnings about unreachable patterns.